messages/mcu_messaging_main.proto

```diff
@@ -796,3 +797,12 @@ message BatteryResetReason
 }
 
 message ConePresent { bool cone_present = 1; }
+
+message McuShutdownScheduled {
+    enum ShutdownReason {
+        UNSPECIFIED = 0;
+        BATTERY_REMOVED = 1;
+    }
+    ShutdownReason shutdown_reason = 1;
+    optional uint32 ms_until_shutdown = 2;
```

Comment:
all fields are `optional` with proto3 so not needed

Reply:
> all fields are optional with proto3
proto3 changes the presence paradigm to "no presence", but as of protoc 3.15, you can add `optional` to fields to opt into "explicit presence", like it was in proto2.

* "No presence" - you can't differentiate between the default value and an absence of a value. Default values are not serialized.
* "Explicit presence" - default values are serialized iff they are set. means that the "presence" of default values can be differentiated.

The reason I chose to use `optional` here is to differentiate between "0 milliseconds until shutdown" and "we never told you how long until shutdown, or maybe this field is deprecated and we don't set it anymore".

Reply:
can you check if nanopb supports it?
does it generate something different when `optional` is used?